activate

```diff
@@ -1,95 +1,218 @@
+    ACTIVATION_ENDPOINT = os.getenv('EOS_PHONE_HOME_ACTIVATE_URL', "https://home.endlessm.com/v1/activate")
+    STATE_DIRECTORY = "/var/lib/eos-phone-home"
+
+    ACTIVATION_VARIABLES = { 'live': 'is_live',
+                             'image': 'image',
+                             'release': 'release',
+                             'vendor': 'vendor',
+                             'product': 'product',
+                             'serial': 'serial' }
+
+    def _get_vendor(self):
+        vendor = None
+
+        try:
+            if os.path.isdir('/sys/class/dmi/id'):
+                with open('/sys/class/dmi/id/sys_vendor') as vendor_file:
+                    vendor = vendor_file.read().strip()
+            else:
+                dt_data = None
+                with open('/proc/device-tree/compatible') as dt_file:
+                    dt_data = dt_file.read().strip()
+
+                dt_fields = dt_data.split(',')
+                vendor = dt_fields[0]
+        except Exception as e:
+            print("Unable to get vendor name!", e, file=sys.stderr)
+
+        print(" - Found vendor:", vendor)
+
+        return vendor.strip()
+
+    def _get_product(self):
+        product = None
+
+        try:
+            if os.path.isdir('/sys/class/dmi/id'):
+                with open('/sys/class/dmi/id/product_name') as product_file:
+                    product = product_file.read().strip()
+            else:
+                dt_data = None
+                with open('/proc/device-tree/compatible') as dt_file:
+                    dt_data = dt_file.read().strip()
+
+                dt_fields = dt_data.split(',')
+                product = dt_fields[1]
+        except Exception as e:
+            print("Unable to get product name!", e, file=sys.stderr)
+
+        print(" - Found product:", product)
+
+        return product.strip()
+
+    def _get_image(self):
+        image = None
+
+        try:
+            image = subprocess.check_output(['attr', '-q', '-g', 'eos-image-version', '/sysroot']).strip()
+        except Exception as e:
+            print("Unable to get /sysroot image name!", e, file=sys.stderr)
+            return None
+
+        print(" - Image:", image)
+
+        if image != "":
+            try:
+                image_root = subprocess.check_output(['attr', '-q', '-g', 'eos-image-version', '/']).strip()
+            except Exception as e:
+                print("Unable to get / image name!", e, file=sys.stderr)
+                return image
+
+            print(" - Image root:", image_root)
+
+            if image_root != "":
+                return image_root
+            else:
+                image
+
+        return None
+
+    def _get_release(self):
+        release = None
+
+        try:
+            release_data = None
+            with open('/etc/os-release') as release_file:
+                release_data = release_file.read()
+
+            release_data = re.split("\n", release_data, flags=re.M)
```

Comment:
Seems a bit ott using a regexp when you can iterate the file with "for line in file" or so - then you can do your splitting/cleaning, as is. But if you do like regexps, you could do one regexp like ^VERSION="?\w*(.+)\w*"?$  to check for VERSION and eat the "" and any whitespace if they are present and give you the value in $1